### DISCUSSION TITLE
How to get the home page location/URL?

### DISCUSSION THREAD

**Discussion by HelloLudger** [datenstrom/community#780] [Ask a question] [answered]
Hello,

I am trying to do something completely unfancy: adding a nav-brand in the navigation, linking to the homepage. In normal mode, it works. I do:

```
<li>
  <a class="nav-brand" href="<?php echo $this->yellow->content->find('/', true)->getUrl() ?>">
    <?php echo $this->yellow->page->getHtml("sitename") ?>
  </a>
</li>
```

But if I am in edit-mode, this code fails with the following error-message:
```
*Fatal error*:  Uncaught Error: Call to a member function getUrl() on null in /...dir.../system/layouts/custom-navigation.html:6
Stack trace:
#0 /...dir.../system/extensions/core.php(669): require()
#1 /...dir.../system/extensions/core.php(351): YellowPage->includeLayout('navigation')
#2 /...dir.../system/layouts/custom-header.html(20): YellowCore->layout('navigation')
#3 /...dir.../system/extensions/core.php(669): require('/hp/ao/ac/sq/ww...')
#4 /...dir.../system/extensions/core.php(351): YellowPage->includeLayout('header')
#5 /...dir.../system/layouts/error.html(1): YellowCore->layout('header')
datenstrom/community#7 /...dir.../system/extensions/core.php(675): require('/hp/ao/ac/sq/ww...')
datenstrom/community#8 /...dir.../system/extensions/core.php(653): YellowPage->includeLayout('error')
#8 /...dir.../system/extensions/core.php(604): YellowPage->parsePageLayout('error')
datenstrom/community#9 /...dir.../system/ in /...dir.../system/layouts/custom-navigation.html on line 6
```

Only workaround I can find is hard coding the base url. But that doesn't seem logically to me. If I release my custom theme as a theme, this should be generated automatically.

I don't understand why I can't access `$this->yellow->content->find('/', true)` if I am in edit mode.

**Comment by schulle4u** [comment, marked as answer]
If you just want a simple link to the homepage, why not use the home link code from `system/layouts/header.html`? In your example: 

`<li><a class="nav-brand" href="<?php echo $this->yellow->page->getBase(true)."/" ?>"><?php echo $this->yellow->page->getHtml("sitename") ?></a></li>`

This will automatically create a link to the home page, appending the appropriate edit URL when in edit mode.
